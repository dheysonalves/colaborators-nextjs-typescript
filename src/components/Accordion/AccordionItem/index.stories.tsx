import { ComponentMeta, ComponentStory } from '@storybook/react'
import AccordionItem from '.'

export default {
	title: 'Composites/Accordion/AccordionItem',
	component: AccordionItem,
} as ComponentMeta<typeof AccordionItem>

export const Default: ComponentStory<typeof AccordionItem> = () => (
	<AccordionItem />
)
