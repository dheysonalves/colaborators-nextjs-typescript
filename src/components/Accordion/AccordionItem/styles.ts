import styled from '@emotion/styled'
import { BaseTypography } from '../../Typography/base.index'

export const ContainerBorder = styled.div`
	border-radius: 8px;
	border: 2px solid #b5f1dd;
	padding: 24px 16px;
`

export const Title = styled(BaseTypography)`
	color: #587169;
	padding-bottom: 6px;
	font-weight: 600;
	font-size: 12px;
	line-height: 16.8px;
`

export const UserTitleName = styled(Title)`
	padding-left: 8px;
	padding-bottom: 0px;
	font-weight: 600;
`

export const Description = styled(Title)`
	padding-bottom: 0px;
	font-weight: 400;
`

export const ButtoHeaderContainer = styled.button`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	outline: none;
	border: none;
	background-color: transparent;
`

export const CustomFlexContainer = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
`

export const GridContainer = styled.div`
	display: grid;
	grid-template-columns: repeat(2, 1fr);
	grid-row-gap: 2rem;
	grid-column-gap: 1rem;
	margin-top: 20px;
`

export const StatusContainer = styled.div`
	margin-top: 2rem;
`

export const UserAvatar = styled.img`
	width: 32px;
	height: 32px;
	border: 4px solid #1dd195;
	border-radius: 80px;
`
