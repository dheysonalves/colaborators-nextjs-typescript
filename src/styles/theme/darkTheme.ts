export const theme = {
	dark: {
		brand: {
			blue: '#034AFD',
		},
		neutral: {
			background: '#F8FAF9',
			light: '#fff',
			black: '#34423D',
			neutral_2: '#F5FAF8',
			neutral_3: '#A3B8B0',
			neutral_5: '#587169',
		},
		primary: {
			default: '#22E0A1',
			green: '#B5F1DD',
		},
		feedback: {
			success: '#B5F1DD',
		},
		tertiary: {
			disabled: '#EAEFED',
		},
	},
}
