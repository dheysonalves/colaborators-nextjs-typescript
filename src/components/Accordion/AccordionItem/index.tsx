import React, { useState } from 'react'
import {
	ContainerBorder,
	UserAvatar,
	CustomFlexContainer,
	Title,
	UserTitleName,
	Description,
	ButtoHeaderContainer,
	GridContainer,
} from './styles'
import { Container } from '../../utils/Flex/flex.styles'
import { MdKeyboardArrowDown, MdKeyboardArrowUp } from 'react-icons/md'
import StatusPin from '../../StatusPin'
import {
	AgentStatusTypes,
	DepartamentRolesTypes,
	NameRolesTypes,
} from '../../../api/@types/general.types'

export interface IAccordionItemProps {
	data: {
		agent_id: number
		name: string
		image: string
		department: DepartamentRolesTypes
		branch: string
		role: NameRolesTypes
		status: AgentStatusTypes
	}
}

const AccordionItem = ({ data }: IAccordionItemProps) => {
	const [toogle, setToogle] = useState(false)

	return (
		<ContainerBorder>
			<Container column>
				<Title tag="h1">Nome Completo</Title>
				<ButtoHeaderContainer
					onClick={() => setToogle(!toogle)}
					aria-label="button-toogle"
				>
					<CustomFlexContainer>
						<UserAvatar src={data.image ?? '/img/brand.svg'} />
						<UserTitleName tag="h1">{data.name}</UserTitleName>
					</CustomFlexContainer>
					{toogle ? (
						<MdKeyboardArrowDown size={24} color="#587169" />
					) : (
						<MdKeyboardArrowUp size={24} color="#587169" />
					)}
				</ButtoHeaderContainer>
				{toogle && (
					<GridContainer>
						<div>
							<Title tag="h1">Departamento</Title>
							<Description tag="p">{data.department}</Description>
						</div>
						<div>
							<Title tag="h1">Cargo</Title>
							<Description tag="p">{data.role}</Description>
						</div>
						<div>
							<Title tag="h1">Unidade</Title>
							<Description tag="p">{data.branch}</Description>
						</div>
						<div>
							<Title tag="h1">Status</Title>
							<StatusPin status={data.status} />
						</div>
					</GridContainer>
				)}
			</Container>
		</ContainerBorder>
	)
}

export default AccordionItem
