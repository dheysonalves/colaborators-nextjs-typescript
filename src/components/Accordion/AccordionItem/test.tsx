import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'

import AccordionItem from '.'
import { theme } from '../../../styles/theme/theme'
import {
	AgentStatusTypes,
	DepartamentRolesTypes,
	NameRolesTypes,
} from 'api/@types/general.types'

const MOCK_DATA = {
	agent_id: 1,
	name: 'Pikachu Soares do Santos Dias',
	branch: 'Quartel General',
	department: DepartamentRolesTypes.ADMINISTRATIVE,
	image: '/img/brand.svg',
	role: NameRolesTypes.ANALYST,
	status: AgentStatusTypes.ACTIVE,
}

const setup = () => {
	const utils = render(
		<ThemeProvider theme={theme}>
			<AccordionItem data={MOCK_DATA} />
		</ThemeProvider>,
	)
	const button = utils.getByLabelText('button-toogle') as HTMLButtonElement

	return {
		button,
		...utils,
	}
}

describe('<AccordionItem />', () => {
	it('should render the name `Pikachu Soares do Santos Dias`', () => {
		const { container, button } = setup()

		fireEvent.click(button)

		expect(
			screen.getByRole('heading', { name: /Nome completo/i }),
		).toBeInTheDocument()

		expect(
			screen.getByText('Pikachu Soares do Santos Dias'),
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the department `Administrativo`', () => {
		const { container, button } = setup()

		fireEvent.click(button)

		expect(
			screen.getByRole('heading', { name: /Departamento/i }),
		).toBeInTheDocument()

		expect(screen.getByText('Administrativo')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the branch `Quartel General`', () => {
		const { container, button } = setup()

		fireEvent.click(button)

		expect(
			screen.getByRole('heading', { name: /Unidade/i }),
		).toBeInTheDocument()

		expect(screen.getByText('Quartel General')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the role `Analista`', () => {
		const { container, button } = setup()

		fireEvent.click(button)

		expect(screen.getByRole('heading', { name: /Cargo/i })).toBeInTheDocument()

		expect(screen.getByText('Analista')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the status `Ativo`', () => {
		const { container, button } = setup()

		fireEvent.click(button)

		expect(screen.getByRole('heading', { name: /Status/i })).toBeInTheDocument()

		expect(screen.getByText('Ativo')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
