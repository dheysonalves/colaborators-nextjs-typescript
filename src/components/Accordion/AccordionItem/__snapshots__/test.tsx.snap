// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> should render the branch \`Quartel General\` 1`] = `
<div
  class="css-hbebh7"
>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-1ccxq8z"
    >
      Nome Completo
    </h1>
    <button
      aria-label="button-toogle"
      class="css-e6mruz"
    >
      <div
        class="css-o34hqt"
      >
        <img
          class="css-1q4j5aa"
          src="/img/brand.svg"
        />
        <h1
          class="css-1fppzxh"
        >
          Pikachu Soares do Santos Dias
        </h1>
      </div>
      <svg
        color="#587169"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </button>
    <div
      class="css-58t2qa"
    >
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Departamento
        </h1>
        <p
          class="css-ud68x3"
        >
          Administrativo
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Cargo
        </h1>
        <p
          class="css-ud68x3"
        >
          Analista
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Unidade
        </h1>
        <p
          class="css-ud68x3"
        >
          Quartel General
        </p>
      </div>
    </div>
    <div
      class="css-nrzz4n"
    >
      <h1
        class="css-1ccxq8z"
      >
        Status
      </h1>
      <div
        class="css-1xl4ljt"
        color="active"
      >
        <h3
          class="css-1rt2vi7"
        >
          Ativo
        </h3>
      </div>
    </div>
  </div>
</div>
`

exports[
	`<AccordionItem /> should render the department \`Administrativo\` 1`
] = `
<div
  class="css-hbebh7"
>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-1ccxq8z"
    >
      Nome Completo
    </h1>
    <button
      aria-label="button-toogle"
      class="css-e6mruz"
    >
      <div
        class="css-o34hqt"
      >
        <img
          class="css-1q4j5aa"
          src="/img/brand.svg"
        />
        <h1
          class="css-1fppzxh"
        >
          Pikachu Soares do Santos Dias
        </h1>
      </div>
      <svg
        color="#587169"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </button>
    <div
      class="css-58t2qa"
    >
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Departamento
        </h1>
        <p
          class="css-ud68x3"
        >
          Administrativo
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Cargo
        </h1>
        <p
          class="css-ud68x3"
        >
          Analista
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Unidade
        </h1>
        <p
          class="css-ud68x3"
        >
          Quartel General
        </p>
      </div>
    </div>
    <div
      class="css-nrzz4n"
    >
      <h1
        class="css-1ccxq8z"
      >
        Status
      </h1>
      <div
        class="css-1xl4ljt"
        color="active"
      >
        <h3
          class="css-1rt2vi7"
        >
          Ativo
        </h3>
      </div>
    </div>
  </div>
</div>
`

exports[
	`<AccordionItem /> should render the name \`Pikachu Soares do Santos Dias\` 1`
] = `
<div
  class="css-hbebh7"
>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-1ccxq8z"
    >
      Nome Completo
    </h1>
    <button
      aria-label="button-toogle"
      class="css-e6mruz"
    >
      <div
        class="css-o34hqt"
      >
        <img
          class="css-1q4j5aa"
          src="/img/brand.svg"
        />
        <h1
          class="css-1fppzxh"
        >
          Pikachu Soares do Santos Dias
        </h1>
      </div>
      <svg
        color="#587169"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </button>
    <div
      class="css-58t2qa"
    >
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Departamento
        </h1>
        <p
          class="css-ud68x3"
        >
          Administrativo
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Cargo
        </h1>
        <p
          class="css-ud68x3"
        >
          Analista
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Unidade
        </h1>
        <p
          class="css-ud68x3"
        >
          Quartel General
        </p>
      </div>
    </div>
    <div
      class="css-nrzz4n"
    >
      <h1
        class="css-1ccxq8z"
      >
        Status
      </h1>
      <div
        class="css-1xl4ljt"
        color="active"
      >
        <h3
          class="css-1rt2vi7"
        >
          Ativo
        </h3>
      </div>
    </div>
  </div>
</div>
`

exports[`<AccordionItem /> should render the role \`Analista\` 1`] = `
<div
  class="css-hbebh7"
>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-1ccxq8z"
    >
      Nome Completo
    </h1>
    <button
      aria-label="button-toogle"
      class="css-e6mruz"
    >
      <div
        class="css-o34hqt"
      >
        <img
          class="css-1q4j5aa"
          src="/img/brand.svg"
        />
        <h1
          class="css-1fppzxh"
        >
          Pikachu Soares do Santos Dias
        </h1>
      </div>
      <svg
        color="#587169"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </button>
    <div
      class="css-58t2qa"
    >
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Departamento
        </h1>
        <p
          class="css-ud68x3"
        >
          Administrativo
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Cargo
        </h1>
        <p
          class="css-ud68x3"
        >
          Analista
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Unidade
        </h1>
        <p
          class="css-ud68x3"
        >
          Quartel General
        </p>
      </div>
    </div>
    <div
      class="css-nrzz4n"
    >
      <h1
        class="css-1ccxq8z"
      >
        Status
      </h1>
      <div
        class="css-1xl4ljt"
        color="active"
      >
        <h3
          class="css-1rt2vi7"
        >
          Ativo
        </h3>
      </div>
    </div>
  </div>
</div>
`

exports[`<AccordionItem /> should render the status \`Ativo\` 1`] = `
<div
  class="css-hbebh7"
>
  <div
    class="css-1s9w40j"
  >
    <h1
      class="css-1ccxq8z"
    >
      Nome Completo
    </h1>
    <button
      aria-label="button-toogle"
      class="css-e6mruz"
    >
      <div
        class="css-o34hqt"
      >
        <img
          class="css-1q4j5aa"
          src="/img/brand.svg"
        />
        <h1
          class="css-1fppzxh"
        >
          Pikachu Soares do Santos Dias
        </h1>
      </div>
      <svg
        color="#587169"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </button>
    <div
      class="css-58t2qa"
    >
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Departamento
        </h1>
        <p
          class="css-ud68x3"
        >
          Administrativo
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Cargo
        </h1>
        <p
          class="css-ud68x3"
        >
          Analista
        </p>
      </div>
      <div>
        <h1
          class="css-1ccxq8z"
        >
          Unidade
        </h1>
        <p
          class="css-ud68x3"
        >
          Quartel General
        </p>
      </div>
    </div>
    <div
      class="css-nrzz4n"
    >
      <h1
        class="css-1ccxq8z"
      >
        Status
      </h1>
      <div
        class="css-1xl4ljt"
        color="active"
      >
        <h3
          class="css-1rt2vi7"
        >
          Ativo
        </h3>
      </div>
    </div>
  </div>
</div>
`
